### PR TITLE
feat!(docs): Updating bzlmod docs

### PR DESCRIPTION
**Pull request by chrislovecnm**
Fixing the bzlmod docs to match latest pip changes.  Also, adding better docs for the release.

### CODE REVIEW

File: .github/workflows/create_archive_and_notes.sh [hunk 1]
```diff
@@ -31,14 +31,20 @@ Add to your \`MODULE.bazel\` file:
 
 \`\`\`starlark
 bazel_dep(name = "rules_python", version = "${TAG}")
+python = use_extension("@rules_python//python:extensions.bzl", "python")
 
-pip = use_extension("@rules_python//python:extensions.bzl", "pip")
+python.toolchain(
+    python_version = "3.10",
+)
+use_repo(python, "python_3_10", "python_aliases")
```
Comment (rickeylev) [inline .github/workflows/create_archive_and_notes.sh:39]: I'd rather omit the toolchain part and python_aliases part and put those into a dedicated doc about using bzlmod. The reason is if and how you should do it depends on whether you're the root module or a submodule (which is further complicated by every submodule is the root module for itself during development).

File: .github/workflows/create_archive_and_notes.sh [hunk 2]
```diff
@@ -31,14 +31,20 @@ Add to your \`MODULE.bazel\` file:
...
 
+pip = use_extension("@rules_python//python:extensions.bzl", "pip")
 pip.parse(
-    name = "pip",
-    requirements_lock = "//:requirements_lock.txt",
+    hub_name = "pip",
```
Comment (rickeylev) [inline .github/workflows/create_archive_and_notes.sh:43]:
```suggestion
    hub_name = "pip",  # This name must be unique across MODULEs
```

**Review by rickeylev — CHANGES_REQUESTED** [review]

File: .github/workflows/create_archive_and_notes.sh (hunk 1 above)
Comment (rickeylev) [reply .github/workflows/create_archive_and_notes.sh:39]: I'd be fine with parring this down to just the `bazel_dep()` call. That's about the minimum you need, and it contains the release-specific information (the version).

Comment (chrislovecnm) [reply .github/workflows/create_archive_and_notes.sh:39]: I am on the fence.  Do many people use the bare minimum?

**Comment by rickeylev** [comment]
Is this PR defunct? We've update the readme quite a bit now

**Comment by chrislovecnm** [comment]
We have already done this work. Closing

**Pull request closed without merging**